### PR TITLE
Mention that softmax might lead to num instability

### PR DESCRIPTION
**Pull request by BenjaminBossan**
Solves #830

By default, skorch uses softmax + `NLLLoss` for classification. This can
lead to numerical instability issues (when logits have large magnitude).
Therefore, mention in the docs how to prevent that issue.

### CODE REVIEW

File: docs/user/quickstart.rst [hunk 1]
```diff
@@ -52,6 +52,12 @@ it on a toy classification dataset using skorch
     net.fit(X, y)
     y_proba = net.predict_proba(X)
 
+.. note::
+
+    The use of ``softmax`` might result in numerical instability. If you have
```
Comment (thomasjpfan) [inline docs/user/quickstart.rst:57]: In general, I think using `CrossEntropyLoss` is the recommended approach. What do you think about updating the example to use `CrossEntropyLoss` and removing the `softmax`?

Comment (BenjaminBossan) [reply docs/user/quickstart.rst]: As always, you come up with the best suggestions. I changed the example accordingly.

**Comment by BenjaminBossan** [comment]
@thomasjpfan could you please take another look?

**Review by thomasjpfan — APPROVED** [review]
LGTM
